eng/common/pipelines/templates/steps/verify-samples.yml

```diff
@@ -5,15 +5,11 @@ parameters:
   - name: ScriptDirectory
     type: string
     default: eng/common/scripts
-  - name: Condition
-    type: boolean
-    default: succeeded()
 
 steps:
   - pwsh: |
       # If the last path segment is an absolute path it will be used entirely.
-      $root = [System.IO.Path]::Combine('$(Build.SourcesDireectory)', 'sdk', '${{ parameters.ServiceDirectory }}')
-      Get-ChildItem $root -Filter *.md -Recurse | ${{ parameters.ScriptDirectory }}\Test-SampleMetadata.ps1 -AllowParentProducts
+      $root = [System.IO.Path]::Combine('$(Build.SourcesDirectory)', 'sdk', '${{ parameters.ServiceDirectory }}')
+      Get-ChildItem $root -Filter *.md -Recurse | ${{ parameters.ScriptDirectory }}/Test-SampleMetadata.ps1 -AllowParentProducts
```

Comment:
Sorry I didn't ask this before but I'm curious why you didn't do the file filtering and path combining inside the script so this is a simple script call. I isn't a big deal but I'm just curious.

Reply:
Because then the script has to do more, which means you have to implement and support more. That's not the PowerShell way. As is, this script - cmdlet, really - can not be run standalone over the whole repo or used for other files. You vary the input coming into the script, and you can use the script in more ways and without reimplementing what other cmdlets already do. I've left this same conceptual feedback for many scripts EngSys has added. 😉

PS is more than a script language - it's a language allowing stringing of cmdlets to pipe objects (as you know, of course). Take advantage of that and you can write more generalized cmdlets that work for more situations by just changing a parameter or varying the input coming into it.

Reply:
While I agree in general I have a tendency to make our scripts be entry points that can be ran locally in isolation to mimic what they would do in CI, as opposed to a general purpose reusable cmdlet. As with everything it is a balancing act and I don't believe most of our script scenarios are really useful for plugging into a larger PS object pipeline.